pkg/estimator/server/framework/plugins/resourcequota/resourcequota.go

```diff
@@ -0,0 +1,361 @@
+	hardResourceList := corev1.ResourceList{}
+	usedResourceList := corev1.ResourceList{}
+	for _, resourceName := range resourceNames {
+		rNameStr := string(resourceName)
+		//skip limits because pb.ReplicaRequirements only support requested resource
+		if strings.HasPrefix(rNameStr, resourceLimitsPrefix) {
+			continue
+		}
+		//requests.cpu is same as cpu
+		//requests.memory is same as memory
+		//we merge them together
+		trimmedResourceName := corev1.ResourceName(strings.TrimPrefix(rNameStr, resourceRequestsPrefix))
+		hardResource, hardResourceOk := rq.Status.Hard[resourceName]
+		usedResource, usedResourceOk := rq.Status.Used[resourceName]
+		if !hardResourceOk || !usedResourceOk {
+			continue
+		}
+		hardResourceList[trimmedResourceName] = hardResource
+		usedResourceList[trimmedResourceName] = usedResource
+	}
+
+	freeResourceList := corev1.ResourceList{}
+	for resourceName, hard := range hardResourceList {
+		if used, ok := usedResourceList[resourceName]; ok {
+			hard.Sub(used)
+			freeResourceList[resourceName] = hard
+		}
+	}
+	return freeResourceList
+}
+
+// resourceNames returns a list of all resource names in the ResourceList
+func resourceNames(resources corev1.ResourceList) []corev1.ResourceName {
+	result := []corev1.ResourceName{}
+	for resourceName := range resources {
+		result = append(result, resourceName)
+	}
+	return result
+}
+
+func getScopeSelectorsFromQuota(quota *corev1.ResourceQuota) []corev1.ScopedResourceSelectorRequirement {
+	selectors := []corev1.ScopedResourceSelectorRequirement{}
+	for _, scope := range quota.Spec.Scopes {
+		selectors = append(selectors, corev1.ScopedResourceSelectorRequirement{
+			ScopeName: scope,
+			Operator:  corev1.ScopeSelectorOpExists})
+	}
+	if quota.Spec.ScopeSelector != nil {
+		selectors = append(selectors, quota.Spec.ScopeSelector.MatchExpressions...)
+	}
+	return selectors
+}
+
+// matchesScope is a function that knows how to evaluate if a pod matches a scope
+// we only support PriorityClass scope now.
+func matchesScope(selector corev1.ScopedResourceSelectorRequirement, priorityClassName string) (bool, error) {
+	switch selector.ScopeName {
+	case corev1.ResourceQuotaScopeTerminating:
+		return false, nil
+	case corev1.ResourceQuotaScopeNotTerminating:
+		return false, nil
+	case corev1.ResourceQuotaScopeBestEffort:
+		return false, nil
+	case corev1.ResourceQuotaScopeNotBestEffort:
+		return false, nil
+	case corev1.ResourceQuotaScopePriorityClass:
+		if selector.Operator == corev1.ScopeSelectorOpExists {
+			// This is just checking for existence of a priorityClass on the pod,
+			// no need to take the overhead of selector parsing/evaluation.
+			return len(priorityClassName) != 0, nil
+		}
+		return matchesSelector(priorityClassName, selector)
+	case corev1.ResourceQuotaScopeCrossNamespacePodAffinity:
+		return false, nil
+	default:
+		klog.Info("selector is not supported", "scope name", selector.ScopeName)
```

Comment:
```suggestion
		// hit here means Kubernetes introduced a new resource quota scope
		klog.Warning("Unrecognized scope name of resource quota", "scope name", selector.ScopeName)
```
Here deserves a warning log, right?